Applications/Tizen_CAPI/capi_func.c

```diff
@@ -299,75 +292,118 @@ int main(int argc, char *argv[]) {
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* add input layer into model */
   status = ml_train_model_add_layer(model, layers[0]);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* create fully connected layer */
   status = ml_train_layer_create(&layers[1], ML_TRAIN_LAYER_TYPE_FC);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* set property for fc layer */
   status = ml_train_layer_set_property(
     layers[1], "unit= 10", "activation=softmax", "bias_initializer=zeros",
     "weight_regularizer=l2norm", "weight_regularizer_constant=0.005",
     "weight_initializer=xavier_uniform", NULL);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* add fc layer into model */
   status = ml_train_model_add_layer(model, layers[1]);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* create optimizer */
   status = ml_train_optimizer_create(&optimizer, ML_TRAIN_OPTIMIZER_TYPE_ADAM);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* set property for optimizer */
   status = ml_train_optimizer_set_property(
     optimizer, "learning_rate=0.0001", "decay_rate=0.96", "decay_steps=1000",
     "beta1=0.9", "beta2=0.9999", "epsilon=1e-7", NULL);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* set optimizer */
   status = ml_train_model_set_optimizer(model, optimizer);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* compile model with cross entropy loss function */
   status = ml_train_model_compile(model, "loss=cross", "batch_size=32", NULL);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* create dataset */
   status = ml_train_dataset_create_with_generator(&dataset, gen_data_train,
                                                   gen_data_val, NULL);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* set property for dataset */
   status = ml_train_dataset_set_property(dataset, "buffer_size=32", NULL);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* set dataset */
   status = ml_train_model_set_dataset(model, dataset);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* train model with data files : epochs = 10 and store model file named
    * "model.bin" */
-  status = ml_train_model_run(model, "epochs=10", "model_file=model.bin", NULL);
-  NN_RETURN_STATUS();
+  status = ml_train_model_run(model, "epochs=10", "save_path=model.bin", NULL);
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   /* delete model */
   status = ml_train_model_destroy(model);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    goto ERROR;
+  }
 
   return 0;
+
+ERROR:
+  printf("run unsuccessful \n");
+
+  ml_train_layer_destroy(layers[0]);
```

Comment:
This isnt right. 
Once layers, optimizer and dataset have been added to the model, these destroy will fail.
lets not call them for all the error cases.

Reply:
I am aware of that. My intention is that if layers/opt/dataset are added to the model, it is essentially noop while before they are set to the model, it will destroy. Correct me, if I am wrong.

Reply:
@zhoonit They are noop from the point of c functions, however it returns error. These applications are going to be references for people to create their own applications. Lets call destroy in proper order.

Reply:
> Instead of using goto, I recommend to add these into macro. NN_RETURN_STATUS()

Thanks for the suggestion. I am afraid using macro is restricted for this case to properly delete objects.

Reply:
Updated